### ISSUE TITLE
Hello, I would like to ask about the choice of the parameter dilated_kernel_size.

### ISSUE THREAD

**Issue by KarsonYu** [K-H-Ismail/Dilated-Convolution-with-Learnable-Spacings-PyTorch#7]
I checked some previous issues where it was mentioned that dilated_kernel_size is related to the receptive field (RF) and currently I have two questions.
(1) For the network module I'm working on [(DiffWave)](https://github.com/lmnt-com/diffwave/blob/master/src/diffwave/model.py) the current parameters are self.dilated_conv = Conv1d(residual_channels, 2 * residual_channels, 3, padding=dilation, dilation= dilation), my understanding here is that if I wish to adjust to Dcls1d, then dilated_kernel_size should be at least 2 * dilation + 1 to be more appropriate, I wonder if this is correct?
2) I'm experimenting with dilated_kernel_size=dilation, but I've noticed that the training is much slower at this point compared to a simple dilation conv, so if I adjust it to 2*dilation+1, will it be even slower? Is it a natural phenomenon for slow training?
I would appreciate if you could give some advice!

**Comment by K-H-Ismail** [comment]
Hello @KarsonYu,
(1) Yes to match the RF of a standard dilated conv, dilated kernel size should be at least (kernel_size-1) * dilation. In your case 2 * dilation. There is another hyper parameter named kernel count (number of elemnts inside the dilated kernel size) that could be tuned as well.

(2) Dcls is slower than conv1d since there is extra computation for positions and their gradients. Also, the implementation of dcls is based on the kernel construction which is more computationally expensive. However with the 1d version, we didn't notice a large drop in throughput as for the 2d or 3d versions. 
If you increase the dilated kernel size you should expect a slower throughput, but it depends also on the size of the input. In the most optimistic setup, moving from dilation to 2*dilation should be slightly slower. Finally running on a recent gpu is advisable.

Sincerely,
@K-H-Ismail

**Comment by KarsonYu** [comment]
> kernel_size

Thank you for your reply, you mentioned the hyperparameter kernel_count,I also did experiments and found that the size of the model changes with the change of kernel_count, so by increasing the kernel_count and decreasing the kernel_size will there be an improvement in the performance of the model or in the speed of training?

**Comment by K-H-Ismail** [comment]
Yes kernel count adds more elements in the kernel thus more learnable parameters. The optimal combination of optimal kernel size kernel count could give best results. Usually we fix kernel count to match the number of parameters of the baseline we want to improve. I did a study on dilated kernel size with respect to classification accruacy in my thesis and found that the bigger the better but it plateaus past certain level (logarithmic gain).
For kernel count it is less clear, as I said you can fix it such as you have less or thz same number of overall parameters as in the baseline.

**Comment by KarsonYu** [comment]
> Yes kernel count adds more elements in the kernel thus more learnable parameters. The optimal combination of optimal kernel size kernel count could give best results. Usually we fix kernel count to match the number of parameters of the baseline we want to improve. I did a study on dilated kernel size with respect to classification accruacy in my thesis and found that the bigger the better but it plateaus past certain level (logarithmic gain). For kernel count it is less clear, as I said you can fix it such as you have less or thz same number of overall parameters as in the baseline.

Thank you for your reply.
